includes/fraud-prevention/class-buyer-fingerprinting-service.php

```diff
@@ -62,12 +62,21 @@ public function hash_data_for_fraud_prevention( string $data ): string {
 	 * @return array An array of hashed data for an order.
 	 */
 	public function get_hashed_data_for_customer( $fingerprint ): array {
+		$order_items_count = WC()->cart ? intval( WC()->cart->get_cart_contents_count() ) : null;
+		$order_id          = isset( $_POST['wcpay_order_id'] ) ? intval( $_POST['wcpay_order_id'], 10 ) : 0; // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
Why aren't we doing nonce verification here? Is it not possible in this context? If it isn't possible, is that why you are casting the value of `$_POST['wcpay_order_id']` to a base 10 integer as a kind of sanitization and validation, just in case?

Reply:
The verification is done on the very most parent method that handles the AJAX request. Doesn't need a second round of validation here because it's not a direct event handler.

Reply:
Got it. Thanks for explaining. I saw the PHPCS ignore line and it wasn't obvious to me at first glance.